File: Ningin/src/physicsManager/colliders/Point.cpp
#include "point.h"

// Define the ZERO constant.
const Point Point::ZERO = Point(0, 0);

void PointFromVector2(Vector2& vec, Point& out)
{
	out = Point(static_cast<int>(vec.x), static_cast<int>(vec.y));
}

void PointFromVector3(Vector3& vec, Point& out)
{
	out = Point(static_cast<int>(vec.x), static_cast<int>(vec.y));
}

void PointToString(Point pt, string& out)
{
	out = format("Point - x: {}, y: {}", pt.x, pt.y);
}

File: NinginCore/physics/Point.cs
namespace NinginCore
{
    public struct Point
    {
        public int x, y;

        public Point(int x = 0, int y = 0)
        {
            this.x = x;
            this.y = y;
        }

        public static Point ToPoint(Vector2 vec)
        {
            InternalCalls.PointFromVector2(ref vec, out Point result);
            return result;
        }

        public static Point ToPoint(Vector3 vec)
        {
            InternalCalls.PointFromVector3(ref vec, out Point result);
            return result;
        }

        public override string ToString()
        {
            InternalCalls.MonoPointToString(ref this, out string result);
            return result;
        }
    }
}
